src/pages/workflows/workflow/common/PermissionsModal.tsx

```diff
@@ -0,0 +1,225 @@
+import { ButtonPrimary, Clickable, Icon, Modal, Select } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { CSSProperties, Dispatch, SetStateAction, useRef, useState } from 'react';
+import { IdContainer, spinnerOverlay } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { TextInput } from 'src/components/input';
+import { getPopupRoot } from 'src/components/popup-utils';
+import { Ajax } from 'src/libs/ajax';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import { FormLabel } from 'src/libs/forms';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import { getTerraUser } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import { append, withBusyState } from 'src/libs/utils';
+import * as Utils from 'src/libs/utils';
+import {
+  publicUser,
+  RawWorkflowsPermissions,
+  WorkflowAccessLevel,
+  WorkflowsPermissions,
+} from 'src/pages/workflows/workflow/workflows-acl-utils';
+
+type WorkflowPermissionsModalProps = {
+  workflowOrNamespace: 'workflow' | 'namespace';
+  namespace: string;
+  name: string;
+  selectedSnapshot: string;
+  setPermissionsModalOpen: (b: boolean) => void;
+};
```

Comment:
I agree. However, since this ticket is simply to introduce the modal and its use for WorkflowPermissions, I didn't want to introduce a new, unused type as part of this PR to avoid scope creep.

Reply:
That makes sense. We can update the type in the namespace permissions modal ticket.